src/main/scala/org/broadinstitute/hail/driver/AnnotateAllelesExpr.scala

```diff
@@ -0,0 +1,96 @@
+package org.broadinstitute.hail.driver
+
+import org.broadinstitute.hail.utils._
+import org.broadinstitute.hail.annotations._
+import org.broadinstitute.hail.expr._
+import org.broadinstitute.hail.methods.Aggregators
+import org.kohsuke.args4j.{Option => Args4jOption}
+
+import scala.collection.mutable
+
+object AnnotateAllelesExpr extends Command {
+
+  class Options extends BaseOptions {
+    @Args4jOption(required = true, name = "-c", aliases = Array("--condition"),
+      usage = "Annotation expression")
+    var condition: String = _
+
+    @Args4jOption(required = false, name = "--propagate-gq", usage = "Propagate GQ instead of computing from PL when splitting alleles")
+    var propagateGQ: Boolean = false
+
+  }
+
+  def newOptions = new Options
+
+  def name = "annotatealleles expr"
+
+  def description = "Annotate alleles programatically"
+
+  def supportsMultiallelic = true
+
+  def requiresVDS = true
+
+  def run(state: State, options: Options): State = {
+    val vds = state.vds
+
+    val cond = options.condition
+    val propagateGQ = options.propagateGQ
+    val isDosage = vds.isDosage
+
+    val (vas2, insertIndex) = vds.vaSignature.insert(TInt, "_aIndex")
+    val (vas3, insertSplit) = vas2.insert(TBoolean, "_wasSplit")
+    val localGlobalAnnotation = vds.globalAnnotation
+
+    val evalva = vds.vaSignature.insert(TInt,"_aIndex")._1.insert(TBoolean,"_wasSplit")._1
+
+    val aggregationST = Map(
+      "global" -> (0, vds.globalSignature),
+      "v" -> (1, TVariant),
+      "va" -> (2, evalva),
+      "g" -> (3, TGenotype),
+      "s" -> (4, TSample),
+      "sa" -> (5, vds.saSignature))
+    val ec = EvalContext(Map(
+      "global" -> (0, vds.globalSignature),
+      "v" -> (1, TVariant),
+      "va" -> (2, evalva),
+      "gs" -> (3, TAggregable(TGenotype, aggregationST))))
+
+    val (parseTypes, fns) = Parser.parseAnnotationArgs(cond, ec, Some(Annotation.VARIANT_HEAD))
```

Comment:
You need to rebase.  The Parser got cleaned up.  This is now `parseAnnotationExprs` and the signature has changed slightly.  That's why the test is failing.